sdk/nodejs/cmd/run/run.ts

```diff
@@ -150,39 +150,17 @@ export function run(
     }
 
     // If this is a typescript project, we'll want to load node-ts.
-    const typeScript: boolean = process.env["PULUMI_NODEJS_TYPESCRIPT"] === "true";
+    const typeScript = process.env["PULUMI_NODEJS_TYPESCRIPT"] === "true";
 
-    // We provide reasonable defaults for many ts options, meaning you don't need to have a tsconfig.json present
-    // if you want to use TypeScript with Pulumi. However, ts-node's default behavior is to walk up from the cwd to
-    // find a tsconfig.json. For us, it's reasonable to say that the "root" of the project is the cwd,
-    // if there's a tsconfig.json file here. Otherwise, just tell ts-node to not load project options at all.
-    // This helps with cases like pulumi/pulumi#1772.
-    const defaultTsConfigPath = "tsconfig.json";
-    const tsConfigPath: string = process.env["PULUMI_NODEJS_TSCONFIG_PATH"] ?? defaultTsConfigPath;
-    const skipProject = !fs.existsSync(tsConfigPath);
-
-    const transpileOnly = (process.env["PULUMI_NODEJS_TRANSPILE_ONLY"] ?? "false") === "true";
-
-    let compilerOptions: object;
-    try {
-        const tsConfigString = fs.readFileSync(tsConfigPath).toString();
-        const tsConfig = parseConfigFileTextToJson(tsConfigPath, tsConfigString).config;
-        compilerOptions = tsConfig["compilerOptions"] ?? {};
-    } catch (e) {
-        compilerOptions = {};
-    }
+    // Users can optionally disable typechecking when using typescript projects 
+    const transpileOnly = process.env["PULUMI_NODEJS_TRANSPILE_ONLY"] === "true";
 
     if (typeScript) {
         tsnode.register({
+            typeCheck: !transpileOnly,
```

Comment:
`typeCheck` is a deprecated option to register replaced by `transpileOnly` which is set on the next line.